### ISSUE TITLE
Chapter 1.4 needs the bugfix ported from Chapter 3.19

### ISSUE THREAD

**Issue by Zireael07** [thebracket/rustrogueliketutorial#22]
I was having a persistent crash in a specific spot in my map, and I thought my map data was corrupt for some reason, until I scrolled down and saw the bugfix....

**Comment by thebracket** [comment]
Good catch! I'll see about getting that back-ported. I'm trying to get the content together for the last section and then planning a big bug-fix run through the whole thing, but that seems important enough to clean up properly. Kinda regretting source code per chapter, it's hard to keep it all in sync!

**Comment by thebracket** [comment]
Just a note that a better fix than the one in 3.14 is to modify the `visibility_system` like this:

```rust
viewshed.visible_tiles = field_of_view(Point::new(pos.x, pos.y), viewshed.range, &*map);
viewshed.visible_tiles.retain(|p| p.x > 0 && p.x < map.width-1 && p.y > 0 && p.y < map.height-1 );
```

I'll try and get that back-ported today.

**Comment by thebracket** [comment]
That should take care of that. I added explanatory text the first time it is used.
